Frontend/Controllers/Projects/TransferDatesController.cs

```diff
@@ -105,70 +103,71 @@ public async Task<IActionResult> TargetDate(string urn, bool returnToPreview = f
 
-            var model = new TransferDatesViewModel {Project = project.Result, ReturnToPreview = returnToPreview};
+            var projectResult = project.Result;
 
-            return View(model);
+            var vm = new TargetDateViewModel
+            {
+                Urn = urn,
+                ReturnToPreview = returnToPreview,
+                TargetDate = new DateViewModel
+                {
+                    Date = DateViewModel.SplitDateIntoDayMonthYear(projectResult.Dates.Target),
+                    UnknownDate = projectResult.Dates.HasTargetDateForTransfer is false
+                }
+            };
+
+            return View(vm);
         }
 
         [HttpPost("target-date")]
         [ActionName("TargetDate")]
-        public async Task<IActionResult> TargetDatePost(string urn, string day, string month, string year,
-            bool returnToPreview = false, bool dateUnknown = false)
+        public async Task<IActionResult> TargetDatePost(TargetDateViewModel vm)
         {
-            var project = await _projectsRepository.GetByUrn(urn);
+            var project = await _projectsRepository.GetByUrn(vm.Urn);
             if (!project.IsValid)
             {
                 return View("ErrorPage", project.Error.ErrorMessage);
             }
 
-            var model = new TransferDatesViewModel {Project = project.Result, ReturnToPreview = returnToPreview};
-            
-            var dateString = DatesHelper.DayMonthYearToDateString(day, month, year);
-
-            model.Project.Dates.Target = dateString;
-            model.Project.Dates.HasTargetDateForTransfer = !dateUnknown;
-
-            if (!string.IsNullOrEmpty(dateString) && !DatesHelper.IsValidDate(dateString))
+            if (!ModelState.IsValid)
             {
-                model.FormErrors.AddError("day", "day", "Enter a valid date");
-                return View(model);
+                return View(vm);
             }
-
-            if (!string.IsNullOrEmpty(model.Project.Dates.Target))
+            
+            var projectResult = project.Result;
+            
+            var validationContext = new ValidationContext<TargetDateViewModel>(vm)
             {
-                if (DatesHelper.SourceDateStringIsGreaterThanToTargetDateString(model.Project.Dates.Htb,
-                    model.Project.Dates.Target) == true)
+                RootContextData =
                 {
-                    model.FormErrors.AddError("day", "day", 
-                        $"The target transfer date must be on or after the Advisory Board date");
-                    return View(model);
+                    ["HtbDate"] = projectResult.Dates.Htb
                 }
-            }
-
-            if (string.IsNullOrEmpty(dateString) && !dateUnknown)
+            };
+            var validator = new TargetDateValidator();
+            var validationResult = await validator.ValidateAsync(validationContext);
+            
+            if (!validationResult.IsValid)
```

Comment:
validationResult.AddToModelState(ModelState, null); - this checks result.IsValid inside the method
Can call !ModelState.IsValid here and remove the one at line 132..